libqtile/bar.py

```diff
@@ -151,14 +157,20 @@ def __init__(self, widgets, size, **config):
         self.queued_draws = 0
 
     def _configure(self, qtile, screen):
-        if self not in [screen.top, screen.bottom]:
-            raise confreader.ConfigError(
-                "Bars must be at the top or the bottom of the screen."
-            )
-        if len([w for w in self.widgets if w.width_type == STRETCH]) > 1:
+        Gap._configure(self, qtile, screen)
+
+        stretches = 0
+        for w in self.widgets:
+            # Executing _test_orientation_compatibility later, for example in
```

Comment:
I'm not sure what you're referring to exactly :) If you're talking about the horizontal/vertical test, if I understand correctly, in `__init__` the bar (and the gap) doesn't know yet where it's been positioned, right? That's also why the previous test was done there too.